substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/NativeImage.java

```diff
@@ -1255,14 +1257,50 @@ protected int buildImage(List<String> javaArgs, LinkedHashSet<Path> bcp, LinkedH
 
         int exitStatus = 1;
         try {
+            Path argsFile = null;
+            Path[] argsFileBox = new Path[1];
+            if (config.useJavaModules()) {
+                /* For Java > 8 we use an argument file to pass the options to the builder */
+                argsFile = Files.createTempFile("native-image", "args");
+                argsFileBox[0] = argsFile;
+                Runtime.getRuntime().addShutdownHook(new Thread() {
+                    @Override
+                    public void run() {
+                        if (argsFileBox[0] != null) {
+                            argsFileBox[0].toFile().delete();
+                        }
+                    }
+                });                
+                argsFile = Files.createTempFile("native-image", "args");
+                argsFile.toFile().deleteOnExit();
+                Files.write(argsFile, (Iterable<String>) command.stream().skip(1).map(NativeImage::quoteFileArg)::iterator);
+                List<String> atCommand = new ArrayList<>();
+                atCommand.add(command.get(0));
+                atCommand.add("@" + argsFile);
+                command = atCommand;
+            }
+            ProcessBuilder pb = new ProcessBuilder();
+            pb.command(command);
             Process p = pb.inheritIO().start();
             exitStatus = p.waitFor();
+            if (exitStatus != 0 && isVerbose() && argsFile != null) {
+                argsFileBox[0] = null;
+                showMessage("Argument file: " + argsFile);
```

Comment:
You can remove the `showMessage` line now.

Reply:
But then how will people find the file as you mentioned before?